add getSwitchCount, getTwoWindingsTransformerCount and getThreeWindin…

MioRtia opened the pull request:
…gsTransformerCount methods

mathbagu commented on iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/NodeBreakerVoltageLevel.java:
```diff
@@ -784,6 +784,11 @@ public void removeAllBuses() {
         }
 
+        @Override
+        public int getSwitchCount() {
```
Are you really sure this view is the good one? NodeBreakerView instead?
By the way, your implementation is not really efficient: you should probably delegate to the calculatedBusBreakerTopology class to avoid a complete iteration.

mathbagu commented on iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/SubstationImpl.java:
```diff
@@ -112,6 +113,11 @@ public TwoWindingsTransformerAdderImpl newTwoWindingsTransformer() {
 
+    @Override
+    public int getTwoWindingsTransformerCount() {
+        return Iterables.size(getTwoWindingsTransformers());
```
I'm not sure this implementation is very efficient. Try this:
```
return voltageLevels.stream()
    .mapToInt(vl -> vl.getConnectableCount(TwoWindingsTransformer.class))
    .sum();
```

mathbagu commented on iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/SubstationImpl.java:
```diff
@@ -129,6 +135,11 @@ public ThreeWindingsTransformerAdderImpl newThreeWindingsTransformer() {
 
+    @Override
+    public int getThreeWindingsTransformerCount() {
+        return Iterables.size(getThreeWindingsTransformers());
```
I'm not sure this implementation is very efficient. Try this:
```
return voltageLevels.stream()
    .mapToInt(vl -> vl.getConnectableCount(TwoWindingsTransformer.class))
    .sum();
```

MioRtia commented:
okay, i fixed the following issues:
- for the getswitchcount method, i added each time the getswitches method is possible and implemented it depending on the cases (it is still available in nodebreakertopology::busbreakerview to allow the option "topologylevel" in XML exporter which changes the topology).
- getTwoWindingsTransformerCount and getThreeWindingsTransformerCount were modified according to your suggestion.
